src/test/java/core/basesyntax/service/RegistrationServiceTest.java

```diff
@@ -0,0 +1,169 @@
+package core.basesyntax.service;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+
+import core.basesyntax.dao.StorageDao;
+import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.db.Storage;
+import core.basesyntax.exception.EmptyFieldException;
+import core.basesyntax.exception.IllegalAgeException;
+import core.basesyntax.exception.InvalidLoginException;
+import core.basesyntax.exception.InvalidPasswordException;
+import core.basesyntax.exception.LoginExistingException;
+import core.basesyntax.model.User;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+public class RegistrationServiceTest {
+    private static final String NULL_TEST_FAILURE_MESSAGE = "Null fields "
+            + "should throw EmptyFieldException!";
+
+    private static final String LOGIN_EXISTS_FAILURE_MESSAGE = "Login already exists in storage "
+            + "should throw LoginExistingException!";
+    private static final String LOGIN_TEST_FAILURE_MESSAGE = "Login less than 6 letters "
+            + "should throw InvalidLoginException!";
+    private static final String PASSWORD_TEST_FAILURE_MESSAGE = "Password less than 6 letters "
+            + "should throw InvalidPasswordException!";
+    private static final String AGE_TEST_FAILURE_MESSAGE = "Age less than 18 "
+            + "or over 122 should throw IllegalAgeException!";
+    private static final String USER_BOB_LOGIN_DEFAULT = "bobjordan";
+    private static final String USER_ALICE_LOGIN_DEFAULT = "alicebaker";
+    private static final String LOGIN_WORSE_CASE = "bob";
+    private static final String USER_PASSWORD_DEFAULT = "password";
+    private static final String PASSWORD_WORSE_CASE = "pass";
+    private static final Integer USER_AGE_DEFAULT = 18;
+    private static final Integer LESS_AGE_WORSE_CASE = 16;
+    private static final Integer OVER_AGE_WORSE_CASE = 210;
+    private static RegistrationService registrationService;
+    private static StorageDao storageDao;
+    private User bob;
+    private User alice;
+
+    @BeforeAll
+    public static void setUp() {
+        registrationService = new RegistrationServiceImpl();
+        storageDao = new StorageDaoImpl();
+    }
+
+    @Test
+    public void registerUser_correctInputData() {
+        bob = new User(USER_BOB_LOGIN_DEFAULT, USER_PASSWORD_DEFAULT, USER_AGE_DEFAULT);
+        alice = new User(USER_ALICE_LOGIN_DEFAULT, USER_PASSWORD_DEFAULT, USER_AGE_DEFAULT);
+        registrationService.register(bob);
+        registrationService.register(alice);
+        User expectedFirstUser = storageDao.get(bob.getLogin());
+        User expectedSecondUser = storageDao.get(alice.getLogin());
+        assertEquals(expectedFirstUser, bob);
+        assertEquals(expectedSecondUser, alice);
+    }
+
+    @Test
+    public void registerDuplicateUser_throwsRuntimeException() {

```

Comment:
[Common mistakes](https://github.com/proxychang/jv-registration-validation/blob/hw-jv-registration-validation/checklist.md#common-mistakes-jv-registration-validation)
- Remember about naming of test methods. There are a lot of ways to name your test methods. The main point is that they should have informative names and be consistent along with other developers in your team. For this task use such convention: `<methodUnderTest>_<state>_<expectedBehavior>`. For example, if we are testing the method `register` with a `null` user's age the test method name should be `register_nullAge_notOk`. `notOk` is because the test expects the register method to throw an exception.